engine/components-api/src/main/java/com/cloud/network/vpc/VpcManager.java

```diff
@@ -122,7 +122,7 @@ public interface VpcManager {
      * @throws InsufficientAddressCapacityException
      * @throws ConcurrentOperationException
      */
-    PublicIp assignSourceNatIpAddressToVpc(Account owner, Vpc vpc) throws InsufficientAddressCapacityException, ConcurrentOperationException;
+    PublicIp assignSourceNatIpAddressToVpc(Account owner, Vpc vpc, Long podId, boolean forNsx) throws InsufficientAddressCapacityException, ConcurrentOperationException;
```

Comment:
how about creating a default parameter and calling an overloaded method.
In addition to that `forNsx` seems very implementation specific; can it be retrieved somehow from configuration and hidden from the service interface?

Reply:
@DaanHoogland thanks I've refactored it to remove the boolean from the service interface